### PR TITLE
Implement Storage Package for Module Data Persistence

### PR DESCRIPTION
**Pull request by dextboy**
## Overview:

This pull request introduces a new storage package along with a `Storage` class dedicated to managing the persistence of module information. The implementation aims to streamline the process of saving and loading module data, ensuring robust data integrity and facilitating easy access to module details across application sessions.

## Features Introduced:

- **Module Data Saving and Loading:** Implements methods to serialize module objects to a file and deserialize them back into application state, enabling persistent storage of module information.
- **Directory and File Management:** Adds utility methods to check for and create necessary directories and files, ensuring a seamless storage experience without manual file management.

## Changes Made:

- Added `moduleList.txt` to `.gitignore` to exclude temporary or local module data from being tracked in Git, addressing potential privacy and relevance concerns.
- Created the `Storage` class within the `seedu.duke.storage` package, encapsulating all storage-related functionalities.
- Implemented methods in `Storage` for file creation, directory checks, data writing, and reading, including error handling to manage exceptions gracefully.

### CODE REVIEW

File: src/main/java/seedu/duke/storage/Storage.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                module.setModuleStatus(true);
+            }
+            if (!moduleGrade.equals("null")) {
+                module.setModuleGrade(moduleGrade);
+            }
+            return module;
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new ModuleException("Module data is corrupted.");
+        }
+    }
```
Comment (sxfoo) [inline src/main/java/seedu/duke/storage/Storage.java:91]: Actually now that I think abt it, JsonManager could as well just have a method that can return `new Module(moduleCode, moduleMC, moduleDate, moduleDescription);`

But anyways, you made do with what u have.

**Review by sxfoo — APPROVED** [review]
Looks good! Some methods are getting too long though

File: src/main/java/seedu/duke/storage/Storage.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            while (input.hasNext()) {
+                String line = input.nextLine();
+                Module module = getModule(line);
+                moduleList.add(module);
+            }
+            input.close();
+        } catch (ModuleException | FileNotFoundException e) {
+            System.out.println("An error occurred while loading modules from file: " + e.getMessage());
+        }
+    }
```
[the remark this answers is not in the document]
Comment (dextboy) [reply src/main/java/seedu/duke/storage/Storage.java:64]: Yup I'll see what I can do, thanks for pointing it out!

File: src/main/java/seedu/duke/storage/Storage.java (hunk 1 above)
Comment (dextboy) [reply src/main/java/seedu/duke/storage/Storage.java:91]: Yup we probably can iterate on it further, probably v2.1 to clean it up!

**Pull request merged**